### ISSUE TITLE
Move Tarski's axioms for geometry into main body

### ISSUE THREAD

**Issue by david-a-wheeler** [metamath/set.mm#823]
I propose that we move into the main body the geometry based on Tarski's axioms as developed by Thierry Arnoux, based on previous work by Scott Fenton but using generic structures. This allows to prove statements for geometry without being constrained on a specific base.  It is currently at the end of set.mm, which is inconvenient for building further work.

set.mm really should have *some* standard way to express geometry.  This seems like a reasonable place to build from.  No doubt there's more to do, but I don't think we need to wait until all geometric proofs are done :-).

It already combines two people's work at least, so it meets the "multiple person" rule for moving out of mathboxes.  The work reuses some work that's still in Scott Fenton's mathbox, so it's more than "copy one section".

@scott-fenton @tirix @nmegill @digama0 - what do you think?

**Comment by tirix** [comment]
That would work for me!
I’d propose to put Geometry in its own part, after part 15, Graph Theory, so that trigonometry is available (for later on..)

**Comment by sctfn** [comment]
Sounds great to me! I’d love to see geometry in the main section.

**Comment by david-a-wheeler** [comment]
@nmegill @digama0 - ok to do?

@tirix - would you be willing to create the pull request to do this, since you understand how to extract this best?

**Comment by nmegill** [comment]
It is ok with me.

**Comment by tirix** [comment]
Yes, I can do that.
Before I move geometry to the main section, I'd like to make the following two clean-ups:

- change the definition of TarskiG so that it does not include the parallel postulate, as David suggested,
- request in the definition of TarskiG that the ` LineG ` slot is defined as stated in ~ tglng

Once this is moved, can I continue developments for geometry directly into the main body of set.mm? What I would really like to achieve is to formalize up to the chapter 14 of [Schwabhaeuser], which equips the geometric line with the structure of an ordered field, but this is very far away!

**Comment by nmegill** [comment]
I think it's ok to continue directly into the main body as long as you think it's something others will find a use for.  For tentative or experimental things you can still use your mathbox.  You can make your own judgment.

**Comment by benjub** [comment]
@tirix When you move stuff to the main part, you can use the header comment I proposed in the discussion group, fixing the "df-tasrkg...", if @nmegill agrees.

Note the paragraph where I emphasized modularity of the axioms and mentioned the subsets which are in my opinion important: in order of increasing specialization, "absolute dimensionless geometry", "Euclidean dimensionless geometry", "planar Euclidean geometry". "Absolute" is here standard for "non-necessarily Euclidean" (this can be stated in the comment of "df-tarkiabs" or whatever you label it.)

Apparently, you can go ahead with Itv --> Seg, and interval -->segment (including in label names).  I think "line segment" can also be simply "segment".

Concerning "lineG", I haven't checked what it entails, but be aware of this: a thing that distinguishes Tarki's from other geometries is that it has a single primitive term: points.  As opposed to e.g. Hilbert geometry, where there are three primitive terms: "points", "lines" and planes". So, while "Line" (why "LineG"?) may of course be defined, I think it should be defined (by the equation analogous to that in ax-tglng, i.e. a disjunction of three betweenness assertions), but not taken as an axiom.

**Comment by nmegill** [comment]
> @tirix When you move stuff to the main part, you can use the header comment I proposed in the discussion group, fixing the "df-tasrkg...", if @nmegill agrees.

I will agree.

**Comment by tirix** [comment]
> (why "LineG"?)

I chose ` LineG ` simply because ` Line ` is [already taken](http://us2.metamath.org:88/mpeuni/df-line2.html)...

**Comment by david-a-wheeler** [comment]
@tirix - I eagerly look forward to the pull request that moves this Tarski's geometry axioms into the main body.

@nmegill - once that happens, I think that we should modify mmrecent.html to note this as news, since it's a major new section.  Something like this (cribbed from the new header):

"The main body of set.mm now has a section on elementary geometry that builds on Tarsky's axioms. Tarski's geometry is a first-order theory with one sort (the "points") and two primitive notions, the ternary predicate of "betweenness" and the quaternary predicate of "congruence".  For generality we treat a Tarski structure as a special kind of extensible structure.

**Comment by tirix** [comment]
I've created the pull request, see #841
If ` LineG ` ends up being used more generally than ` Line ` , I think we can still rename it.
Similarly, I have followed very closely the naming of Scott's theorems, simply adding "tg" before.
Since the main part of set.mm is to be developed further, it might be good to swap this convention, and have the main part use names without prefixes, while adding prefixes to Scott's (line "ee" for example)

**Comment by david-a-wheeler** [comment]
@tirix .. yes, the theorems in the main body should not have prefixes, since in some sense they are the "real thing".

**Comment by david-a-wheeler** [comment]
I'm not sure that I can really comment on the more technical parts, but we really need geometry section in the main body, and we agreed that the Tarski axioms with extendable structures were the way to do it. It seems to me you've met that requirement. Unless there is some objection, I think we should merge this.

**Comment by david-a-wheeler** [comment]
I hope that @scott-fenton will be willing to tweak some of his work to match this version of Tarski's axioms.  After all, it's all based on his work originally.

**Comment by benjub** [comment]
Why has the title been changed from "Elementary geometry (Tarski's axioms)" to "Geometry"?  This is at best less descriptive, but actually I would say this is simply misleading.

**Comment by tirix** [comment]
That's being ambitious! 😉
Just like the "graph theory" part does not contain all of graph theory, this part does not contain all of geometry. I believed parts would be better named after the [mathematical topics](https://en.wikipedia.org/wiki/Mathematics_Subject_Classification) they belong to, but you have a point.

My last efforts have been to introduce ` LineG ` as a structure slot, so that it can be a primitive in other systems of axioms like Hilberts.  In any event, further developments shall be made to prove Hilbert's axioms, so we could later claim this is more than just Tarskian geometry.

Until then, I shall change the title back.

**Comment by benjub** [comment]
@tirix : to be clear: this is not to belittle your work at all, whatsoever.  My area of expertise is differential geometry, and I also studied algebraic geometry, so maybe I overreacted a little!  Maybe "Elementary Geometry" is better.

**Comment by digama0** [comment]
I think calling it Geometry, or Elementary Geometry, is good. The intent is to specify "what part of maths goes here" rather than "what is currently in this section". If you've only got as far as the axiomatization then that would be the first and only chapter in the part.

**Comment by david-a-wheeler** [comment]
It's true that it will *currently* only contain Tarski's axioms for geometry, but if we add more ways to deal with geometry (other than deprecated work), then I presume they would go in the same major section.  Also, "Geometry" is a wonderfully simple and clear title, I think it's better that way.  So I agree with @digama0 , let's call it "Geometry" or "Elementary Geometry".

**Comment by benjub** [comment]
Let's go for "Elementary Geometry".  I like this axiomatic approach very
much, but one should be aware of two things:

* It is very marginal in the mathematical "edifice", compared to, say,
algebraic geometry or differential geometry (or even combinatorial
geometry).  This is why I said that the title "Geometry" would be
misleading.

* Second, if/when these more mainstream topics get formalized in metamath,
they will not arise as developments of this elementary geometry (for
instance, algebraic geometry will sprout from commutative algebra, and
differential geometry from topology, mainly). (This is to answer the
"currently" in David's message).

**Comment by tirix** [comment]
The new title is in the last pull request #844!

**Comment by david-a-wheeler** [comment]
@tirix - I'm eagerly waiting for the pull request to move Tarski geometry into the main body.  Are you still working on that?

**Comment by tirix** [comment]
@david-a-wheeler it is actually already there since last week, pull request #841 !

**Comment by david-a-wheeler** [comment]
@tirix - As far as I can tell the new geometry section is still not in the main body.  Can you submit a change to fix that? Or....

@nmegill - am I misunderstanding something?

I just looked at the current version of devel, commit e311e87ca8362fb662093887bc0831a60b97654d dated Wed May 15 08:02:21. The deprecated material section begins on line 358469, and the mathboxes begin on line 393602.

The new definition of "ELEMENTARY GEOMETRY" begins on line 595821, which is WAY after the beginning of the deprecated material section and mathboxes.  To be part of the main body I believe it needs to be *before* the deprecated material section.

Thanks!

~~~~
$( Begin $[ set-tarskigeom.mm $] $)
$(
#############################################################################
                      ELEMENTARY GEOMETRY
#############################################################################

This part develops elementary geometry based on Tarski's axioms, following
[Schwabhauser].  Tarski's geometry is a first-order theory with one sort,
~~~~

**Comment by tirix** [comment]
From what I can see, in that commit, the geometry part begins on line 343313.
In the [theorem list (table of contents)](http://us2.metamath.org/mpeuni/mmtheorems.html), it appears as part 15, just before _graph theory_ (part 16), and just after _basic real and complex functions_ (part 14).
That's (towards the end of) the main part.

**Comment by david-a-wheeler** [comment]
You're right!  I don't know what happened there, but all looks good.  I wonder if I accidentally opened the wrong version of set.mm.

Thanks for clarifying things.

In that case, I think we can close this issue.

@nmegill - I think adding Tarski's axioms for geometry deserves an entry in the Metamath "recent news items" in mmrecent.html.

**Comment by nmegill** [comment]
> @nmegill - I think adding Tarski's axioms for geometry deserves an entry in the Metamath "recent news items" in mmrecent.html.

OK.  Could you propose here what the entry should say?  Then I will insert it.

**Comment by david-a-wheeler** [comment]
Okay, how about this?

> Set.mm now has a major new section on <a href="mpeuni/mmtheorems.html#dtl:15">elementary geometry</a>.  This begins with definitions that implement Tarski's axioms of geometry (including concepts such as conguence and betweenness).  This uses set.mm's extensible structures, making them easier to use for many circumstances. The section then connects Tarski geometry to Hilbert spaces, in particular with geometry in the complex plane and geometry in Euclidean places. Most of the work in this section is due to Thierry Arnoux, with earlier work by Mario Carneiro and Scott Fenton.

**Comment by nmegill** [comment]
Thank you.  It is updated on us2.   The news item is a little longer than typical, but I posted it as is.

I made the link "mmtheorems.html#cstrkg" where cstrkg is the first $a or $p below the section header. This is the convention for linking to section headers containing text, so that the link will not depend on the section numbering.  I described this here:
https://groups.google.com/d/msg/metamath/EhV8FR0LOhc/onJKCF0FBwAJ
